### PR TITLE
[5985] correct count of contacts in flow

### PR DESCRIPTION
**Pull request by SilviuOpenline**
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Additional context

----

> [!IMPORTANT]
> Corrects contact count display in `Header.tsx` by using `store.ui.searchCount` instead of `flow?.value?.participants?.length`.
> 
>   - **Behavior**:
>     - Corrects contact count display in `Header` component of `Header.tsx`.
>     - Replaces `flow?.value?.participants?.length` with `store.ui.searchCount` for accurate contact count.
>   - **UI**:
>     - Updates contact count logic to use `store.ui.searchCount` for consistency with UI state.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=customeros%2Fcustomeros&utm_source=github&utm_medium=referral)<sup> for 049241fa501af4f1b3b9908d22aa2cef40846753. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by Kasia9090 — APPROVED** [review]

File: packages/apps/frontera/src/routes/flow-editor/src/Header.tsx [hunk 1]
```diff
@@ -145,10 +145,8 @@ export const Header = observer(
                 <>
                   <ChevronRight className='text-gray-400' />
                   <span className='font-medium cursor-default'>
-                    {`${flow?.value?.participants?.length} ${
-                      flow?.value?.participants?.length === 1
-                        ? 'Contact'
-                        : 'Contacts'
+                    {`${store.ui.searchCount} ${
+                      store.ui.searchCount === 1 ? 'Contact' : 'Contacts'
```
Comment (acalinica) [inline packages/apps/frontera/src/routes/flow-editor/src/Header.tsx:149]: searchCount is used for something else. The source of truth for this count needs to be flow.value.participants.length.

Comment (SilviuOpenline) [reply packages/apps/frontera/src/routes/flow-editor/src/Header.tsx:149]: @acalinica the participants.length includes the archived contacts

Comment (acalinica) [reply packages/apps/frontera/src/routes/flow-editor/src/Header.tsx:149]: That still does not justify using store.ui.searchCount for this. Store.ui.searchCount can be set from many other places in the app that are unrelated with this logic, hence it's not safe to use it like this.

Comment (umihai29) [reply packages/apps/frontera/src/routes/flow-editor/src/Header.tsx:149]: just filter the participants to not include the one that are archived

Comment (acalinica) [reply packages/apps/frontera/src/routes/flow-editor/src/Header.tsx:149]: The right thing to do here is to change the query that returns this count to not sum up the archived ones.

Comment (acalinica) [reply packages/apps/frontera/src/routes/flow-editor/src/Header.tsx:149]: Or just filter the participants that are archived and then get the length.

**Pull request closed without merging**